src/server/game/Server/WorldSession.h

```diff
@@ -348,8 +348,22 @@ class WorldSession
     void SendPacket(WorldPacket const* packet);
     void SendPetNameInvalid(uint32 error, std::string const& name, DeclinedName* declinedName);
     void SendPartyResult(PartyOperation operation, std::string const& member, PartyResult res, uint32 val = 0);
-    void SendAreaTriggerMessage(const char* Text, ...) ATTR_PRINTF(2, 3);
-    void SendAreaTriggerMessage(uint32 entry, ...);
+
+    void SendAreaTriggerMessage(std::string_view str);
+
+    template<typename... Args>
+    void SendAreaTriggerMessage(char const* fmt, Args&&... args)
+    {
+        if (!m_playerLoading)
+            SendAreaTriggerMessage(Acore::StringFormat(fmt, std::forward<Args>(args)...));
+    }
+    template<typename... Args>
+    void SendAreaTriggerMessage(uint32 strId, Args&&... args)
+    {
+        if (!m_playerLoading)
```

Comment:
Why not hassession

Reply:
@Kitzunu, HasSession() is available via the ChatHandler. I can use that if it's requested. 
I went with a variable that is available in the WorldSession itself.

Reply:
true, my bad